File: practice/algorithms/implementation/library_fine/LibraryFine.java
import java.util.Scanner;

public class LibraryFine
{
    public static void main(String[] args)
    {
        Scanner stdin = new Scanner(System.in);
        int actualDay = stdin.nextInt();
        int actualMonth = stdin.nextInt();
        int actualYear = stdin.nextInt();
        int expectedDay = stdin.nextInt();
        int expectedMonth = stdin.nextInt();
        int expectedYear = stdin.nextInt();

        int yearDiff = actualYear - expectedYear;
        int monthDiff = actualMonth - expectedMonth;
        int dayDiff = actualDay - expectedDay;

        System.out.println(fee(yearDiff, monthDiff, dayDiff));
    }

    private static int fee(int year, int month, int day)
    {
        if(year > 0) return 10000;
        if(year == 0 && month > 0) return month * 500;
        if(year == 0 && month == 0 && day > 0) return day * 15;
        return 0;
    }
}

File: practice/algorithms/implementation/library_fine/library_fine.rb
(actual_day, actual_month, actual_year) = gets.to_s.split(" ").map { |s| s.to_i }
(expected_day, expected_month, expected_year) = gets.to_s.split(" ").map { |s| s.to_i }

year_diff = actual_year - expected_year
month_diff = actual_month - expected_month
day_diff = actual_day - expected_day

def fee(year, month, day)
  if year > 0
    return 10000
  end
  if year == 0 && month > 0
    return month * 500
  end
  if year == 0 && month == 0 && day > 0
    return day * 15
  end
  return 0
end

puts fee(year_diff, month_diff, day_diff)